### PR TITLE
TextFormatter: Support replacing \n without trusted

### PR DESCRIPTION
**Pull request by KrisXV**

### CODE REVIEW

File: test/server/chat.js [hunk 1]
```diff
@@ -88,5 +88,17 @@ describe('Chat', function () {
 			Chat.formatText(`:)`, true),
 			`:)`
 		);
+		assert.equal(
+			Chat.formatText(`a\nb\nc`),
+			`a\nb\nc`
+		);
+		assert.equal(
```
Comment (AnnikaCodes) [inline test/server/chat.js:95]: These should all be `strictEqual` but that is for another PR

**Comment by AnnikaCodes** [comment]
The changes to my plugin are fine. Are there places where we want to format text with the trusted flag without replacing `\n`? I wouldn't expect there to be.

File: test/server/chat.js (hunk 1 above)
Comment (KrisXV) [reply test/server/chat.js:95]: I'm aware

**Comment by KrisXV** [comment]
> The changes to my plugin are fine. Are there places where we want to format text with the trusted flag without replacing `\n`? I wouldn't expect there to be.

It auto replaces `\n` if `isTrusted === true`

File: test/server/chat.js (hunk 1 above)
Comment (Zarel) [reply test/server/chat.js:95]: NO THEY SHOULDN'T

Notice the top of this file:

```js
const assert = require('assert').strict;
```

TypeScript marking these as wrong is a TypeScript bug.

Comment (Zarel) [reply test/server/chat.js:95]: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/node/assert.d.ts

lol, this was fixed 3 days ago.

Comment (KrisXV) [reply test/server/chat.js:95]: It's not typescript marking these wrong, `strict` still works but is is deprecated

**Comment by Zarel** [comment]
I think the confusion is because the PR is called "Allow" rather than "Support", which suggests there are permission changes rather than just API changes.

**Comment by Zarel** [comment]
Anyway thanks, and please do NOT replace `equal` with `strictEqual` anywhere in the codebase.

File: test/server/chat.js (hunk 1 above)
Comment (Zarel) [reply test/server/chat.js:95]: What do you mean?

Comment (Zarel) [reply test/server/chat.js:95]: `require('assert').strict` is not only not deprecated, `require('assert')` _is_ deprecated...

Comment (AnnikaCodes) [reply test/server/chat.js:95]: Oh, sorry, I didn't know that.

**Pull request merged**